scripts/mtgfetch.coffee

```diff
@@ -0,0 +1,89 @@
+# Description:
+#   Fetches MTG image and card information. 
+#	These names need to be an exact match. 
+#
+#	I love me some apostrophes. This is v0.01
+#
+# Configuration:
+#   None
+#
+# Commands:
+#	robort mtgme <card>
+#
+# Future ideas:
+#	robort mtgme -t <card>
+#	This only prints the card text, no image. 
+#
+# Author:
+#   mickey
+
+module.exports = (robot) ->
+  robot.respond /mtgme (.*)/i, (msg) ->
+  	# Get the card
+  	cardname = msg.match[1]
+  	
+  	# Check for apostrophe
+  	hasApostrophe = cardname.indexOf('\'') >= 0
+  	if hasApostrophe 
+  		robot.http("""https://api.magicthegathering.io/v1/cards?name=#{cardname}""")
+	  		.get() (err, res, body) -> 
+	  			json = JSON.parse(body)
+	  			numCards = Object.keys(json.cards).length
+
+	  			# Check if there isn't an exact match
+	  			# Do a little cheating here
+	  			# Make a map for the card name
+	  			sameArray = {}
+	  			sameArray[cardname] = 0
+	  			for card in json.cards
+	  				sameArray[card.name]++
+
+	  			# Check if there are multiple cards that do not include promo or gameday
+	  			if(numCards > 1 && sameArray[cardname] != numCards)
+	  				msg.send '''
+	  					Please be more specific:
+	  				'''
+	  				msg.send card.name for card in json.cards
+	  			# If there is an exact match for a card name
+	  			else if (sameArray[cardname] == numCards)					
```

Comment:
This is a mapping value for the cardname. I'm comparing the count of the cardname instance to the total number of cards. 

There are instances such as Languish or Gideon's Phalanx that have multiple printings of the same card. In this case, I need to not show "Please be more specific" if the number cards found is greater than 1 but they are all identically named.